gossip_test.go

```diff
@@ -22,7 +24,7 @@ var _ gossipConnection = &mockGossipConnection{}
 
 func newTestRouter(name string) *Router {
 	peerName, _ := PeerNameFromString(name)
-	router := NewRouter(Config{}, peerName, "nick", nil)
+	router := NewRouter(Config{}, peerName, "nick", nil, log.New(ioutil.Discard, "", 0))
```

Comment:
I am optimizing for quiet logs in the happy path. In the case of failures this has worked well for me in the past:

1. Test fails
2. Modify test, add logger
3. Run test again
4. Identify failure

Is that insufficient here?